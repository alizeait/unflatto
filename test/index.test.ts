const mock = require("../mock");
const { unflatto } = require("../src");

describe("unflatto", () => {
  test("should unflatten correctly", () => {
    expect(unflatto(mock)).toMatchSnapshot();
  });
  test("should unflatten with correct seperator", () => {
    const data = {
      "key1-a": 2,
      "key2-b": 3,
    };
    expect(unflatto(data, "-")).toEqual({ key1: { a: 2 }, key2: { b: 3 } });
    expect(unflatto(mock)).toMatchSnapshot();
  });
  test("should keep falsy values", () => {
    const data = {
      "key1.a": false,
      "key2.b": null,
      "key3.c": 0,
      "key4.d.e": "",
      "key5.f.g": undefined,
      "key6.h.i": NaN,
    };
    expect(unflatto(data)).toEqual({
      key1: { a: false },
      key2: { b: null },
      key3: { c: 0 },
      key4: { d: { e: "" } },
      key5: { f: { g: undefined } },
      key6: { h: { i: NaN } },
    });
  });
  test("should unflatten arrays correctly", () => {
    const data = {
      "0": false,
      "1": null,
      "2": 0,
      "3": "",
      "4": undefined,
      "5": NaN,
    };
    expect(unflatto(data)).toEqual([false, null, 0, "", undefined, NaN]);
    expect(unflatto(data)).toMatchSnapshot();
  });
});
