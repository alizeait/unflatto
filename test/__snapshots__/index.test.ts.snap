// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unflatto should unflatten arrays correctly 1`] = `
Array [
  false,
  null,
  0,
  "",
  undefined,
  NaN,
]
`;

exports[`unflatto should unflatten correctly 1`] = `
Object {
  "key1": Object {
    "keyA": "valueI",
  },
  "key2": Object {
    "keyB": "valueII",
  },
  "key3": Object {
    "a": Object {
      "b": Object {
        "c": 2,
      },
    },
  },
  "key4": Array [],
  "key5": Object {
    "a": Array [
      "value1",
      "value2",
      Object {
        "key1": Object {
          "keyA": "valueI",
          "keyB": Array [
            1,
            2,
            3,
            4,
          ],
        },
      },
    ],
    "b": null,
    "c": undefined,
  },
  "key6": Object {},
}
`;

exports[`unflatto should unflatten with correct seperator 1`] = `
Object {
  "key1": Object {
    "keyA": "valueI",
  },
  "key2": Object {
    "keyB": "valueII",
  },
  "key3": Object {
    "a": Object {
      "b": Object {
        "c": 2,
      },
    },
  },
  "key4": Array [],
  "key5": Object {
    "a": Array [
      "value1",
      "value2",
      Object {
        "key1": Object {
          "keyA": "valueI",
          "keyB": Array [
            1,
            2,
            3,
            4,
          ],
        },
      },
    ],
    "b": null,
    "c": undefined,
  },
  "key6": Object {},
}
`;
